File: Lecture 11/Python Code/Tomato/models/PickupOrder.py
from models.order import Order

class PickupOrder(Order):
    def __init__(self, user, restaurant, items, pickup_time, order_type="Pickup"):
        super().__init__(user, restaurant, items, order_type, pickup_time)
        self.pickup_time = pickup_time

    def get_pickup_time(self):
        return self.pickup_time

    def set_pickup_time(self, time):
        self.pickup_time = time

File: Lecture 11/Java Code/Tomato/models/PickupOrder.js
import { Order } from './Order.js';

export class PickupOrder extends Order {
    constructor() {
        super();
        this.restaurantAddress = "";
    }

    getType() {
        return "Pickup";
    }

    setRestaurantAddress(addr) {
        this.restaurantAddress = addr;
    }

    getRestaurantAddress() {
        return this.restaurantAddress;
    }
}
